nestkernel/event_delivery_manager.cpp

```diff
@@ -356,7 +383,21 @@ EventDeliveryManager::gather_spike_data_( const thread tid,
       // Needs to be called /after/ set_end_and_invalid_markers_.
       set_complete_marker_spike_data_( assigned_ranks, send_buffer_position, send_buffer );
 #pragma omp barrier
+      // TODO: Discuss, if the barrier above should be better placed after if-clause
+    }
+
+#ifdef TIMER_DETAILED
+#pragma omp barrier
```

Comment:
I guess this barrier is placed here in order to guarantee that all threads take the same amount of time and tid = 0 represents all other threads. But I think every new barrier would introduce significant synchronization overhead and distort the overall measurement. This would be especially severe if we used these timers on architectures with high core counts allowing a huge number of threads.

Reply:
Agreed. Now removed barrier.